libtiledbsoma/src/soma/soma_group.cc

```diff
@@ -110,6 +113,31 @@ SOMAGroup::SOMAGroup(
     , uri_(util::rstrip_uri(group->uri()))
     , group_(group)
     , timestamp_(timestamp) {
+    switch (group_->query_type()) {
+        case TILEDB_READ:
+            soma_mode_ = OpenMode::soma_read;
+            break;
+        case TILEDB_WRITE:
+            soma_mode_ = OpenMode::soma_write;
+            break;
+        case TILEDB_MODIFY_EXCLUSIVE:  // Not supported in SOMA
```

Comment:
the only difference between this case and the default case is the exception type.  Is there a reason to treat this mode differently (e.g., InvalidArgument error vs TileDBSOMAError?)